applications/GeoMechanicsApplication/custom_conditions/surface_normal_load_3D_diff_order_condition.cpp

```diff
@@ -36,37 +37,29 @@ Condition::Pointer SurfaceNormalLoad3DDiffOrderCondition::Create(IndexType NewId
     return Condition::Pointer(new SurfaceNormalLoad3DDiffOrderCondition(NewId, GetGeometry().Create(ThisNodes), pProperties));
 }
 
-//----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
-void SurfaceNormalLoad3DDiffOrderCondition::
-CalculateConditionVector(ConditionVariables& rVariables, unsigned int PointNumber)
+void SurfaceNormalLoad3DDiffOrderCondition::CalculateConditionVector(ConditionVariables& rVariables,
+                                                                     unsigned int PointNumber)
 {
     KRATOS_TRY
 
-    double NormalVector[3];
-
-    NormalVector[0] = rVariables.JContainer[PointNumber](1,0) * rVariables.JContainer[PointNumber](2,1) -
-                      rVariables.JContainer[PointNumber](2,0) * rVariables.JContainer[PointNumber](1,1);
-
-    NormalVector[1] = rVariables.JContainer[PointNumber](2,0) * rVariables.JContainer[PointNumber](0,1) -
-                      rVariables.JContainer[PointNumber](0,0) * rVariables.JContainer[PointNumber](2,1);
+    Vector normal_vector(3);
+    MathUtils<double>::CrossProduct(normal_vector, column(rVariables.JContainer[PointNumber], 0),
+                                    column(rVariables.JContainer[PointNumber], 1));
 
-    NormalVector[2] = rVariables.JContainer[PointNumber](0,0) * rVariables.JContainer[PointNumber](1,1) -
-                      rVariables.JContainer[PointNumber](1,0) * rVariables.JContainer[PointNumber](0,1);
-    
-    const GeometryType& rGeom = GetGeometry();
-    const SizeType NumUNodes = rGeom.PointsNumber();
-    double NormalStress = 0.0;
-    rVariables.ConditionVector.resize(3,false);
+    const auto& r_geometry                   = GetGeometry();
+    const auto  number_of_displacement_nodes = r_geometry.PointsNumber();
 
-    for ( SizeType i = 0; i < NumUNodes; ++i ) {
-        NormalStress += rVariables.Nu[i]*rGeom[i].FastGetSolutionStepValue(NORMAL_CONTACT_STRESS);
-    }
+    Vector normal_stresses(number_of_displacement_nodes);
```

Comment:
A very minor thing: you could eliminate variable `number_of_displacement_nodes`, since it is only used to give vector `normal_stresses` the correct size:
```suggestion
    Vector normal_stresses(r_geometry.PointsNumber());
```

Reply:
Done